Firestore/Swift/Tests/Codable/CodableTimestampTests.swift

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+import FirebaseFirestoreSwift
+import Foundation
+import XCTest
+
+class CodableTimestampTests: XCTestCase {
+  func testTimestampEncodes() {
+    let timestamp = Timestamp(seconds: 37, nanoseconds: 123)
+
+    let jsonData = try! JSONEncoder().encode(timestamp)
+    let json = String(data: jsonData, encoding: .utf8)!
+
+    // The ordering of attributes in the JSON output is not guaranteed, nor is the rounding of
+    // the values so just verify that each required property is present and that the value
+    // starts as expected.
+    XCTAssert(json.contains("\"seconds\":37"))
+    XCTAssert(json.contains("\"nanoseconds\":123"))
```

Comment:
I wonder, would encoding, decoding and comparing the original and the decoded objects be a simpler way to tests `Codable`? So it will be like:
```
let jsonData = try JSONEncoder().encode(timestamp)
let decodedTimestamp = try JSONDecoder().decode(Timestamp.self, from: jsonData)

XCTAssertEqual(timestamp.seconds, decodedTimestamp.seconds)
...
```
It seams to me that at the end of the day we are interested in ability  to encode/decode rather than in a particular format. Feel free to correct me if I am missing anything.

Reply:
Round trips are definitely interesting (and we do that elsewhere), but since we're essentially promising to external users that the format is stable we need to verify that we don't change the external format as well.

For example, we could rename `nanoseconds` to `nanos` and a round-trip based test would succeed. This test would fail (and fail correctly).

Basically the JSON format we're generating here is an API surface area and we need to validate that we're not breaking it.

Reply:
Ah, that makes sense, thank you for explaining!